editor/src/components/canvas/controls/observer-hooks.ts

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { fastForEach } from '../../../core/shared/utils'
+import * as EP from '../../../core/shared/element-path'
+import { CanvasRectangle, windowPoint } from '../../../core/shared/math-utils'
+import { windowToCanvasCoordinatesGlobal } from '../dom-lookup'
+
+function attachObserverToScrollingElement(
+  observerRef: React.MutableRefObject<MutationObserver | null | undefined>,
+) {
+  // TODO FIND SOMETHING NICER
```

Comment:
Something nicer than what?

Reply:
Oh, than the old `node-connectors` div. Yes. Please create an element specifically for this so that we're not attaching to something unrelated that may well change behaviour later.